include/jemalloc/internal/extent_inlines.h

```diff
@@ -211,6 +220,12 @@ extent_szind_set(extent_t *extent, szind_t szind) {
 }
 
 JEMALLOC_INLINE void
+extent_nfree_set(extent_t *extent, unsigned nfree) {
+	extent->e_bits = (extent->e_bits & ~EXTENT_BITS_NFREE_MASK) |
```

Comment:
I left that assertion out because it would require strict ordering between setting ```slab``` to true and initializing ```nfree```, but now that I look more closely at current setter callsites, this seems like a reasonable restriction.